unit/analyses/variable-sensitivity/value_set/abstract_value.cpp

```diff
@@ -359,3 +361,117 @@ TEST_CASE(
 
   REQUIRE(value_set.to_constant() == value);
 }
+
+static abstract_environmentt get_value_set_abstract_environment()
+{
+  vsd_configt config;
+  config.advanced_sensitivities.new_value_set = true;
+  config.context_tracking.data_dependency_context = false;
+  config.context_tracking.last_write_context = false;
+  variable_sensitivity_object_factoryt::instance().set_options(config);
+  return abstract_environmentt{};
+}
+
+TEST_CASE("Eval on an constant gives us that constant", VALUE_SET_TEST_TAGS)
+{
+  const auto environment = get_value_set_abstract_environment();
+  namespacet ns{symbol_tablet{}};
+  const auto zero = from_integer(0, signedbv_typet{32});
+  const auto eval_result = environment.eval(zero, ns);
+  REQUIRE(eval_result != nullptr);
+  REQUIRE(!eval_result->is_top());
+  REQUIRE(!eval_result->is_bottom());
+  const auto eval_result_as_value_set =
+    std::dynamic_pointer_cast<const value_set_abstract_valuet>(
+      eval_result->unwrap_context());
+  REQUIRE(eval_result_as_value_set != nullptr);
+  REQUIRE(eval_result_as_value_set->get_values().size() == 1);
+  REQUIRE_THAT(eval_result_as_value_set->get_values(), ContainsAllOf{zero});
+}
+
+TEST_CASE(
+  "Eval on a plus expression with constant operands gives us the result of "
+  "that plus",
+  VALUE_SET_TEST_TAGS)
+{
+  const auto environment = get_value_set_abstract_environment();
+  namespacet ns{symbol_tablet{}};
+  const auto s32_type = signedbv_typet{32};
+  const auto three = from_integer(3, s32_type);
+  const auto five = from_integer(5, s32_type);
+  const auto three_plus_five = plus_exprt{three, five};
+
+  auto eval_result = environment.eval(three_plus_five, ns);
+  REQUIRE(eval_result != nullptr);
+  REQUIRE(!eval_result->is_top());
+  REQUIRE(!eval_result->is_bottom());
```

Comment:
This is failing apparently? Strange, I could swear I remember this working.

Reply:
I'm investigate this now. Back soon :)

Reply:
They pass on your original branch. I'm trying to investigate further ...

Reply:
Change in the abstract_environment default constructor 

This branch has 
`abstract_environmentt::abstract_environmentt() : bottom(true) { }`
while the branch this change was originally targetting has 
`abstract_environmentt::abstract_environmentt() : bottom(false) { }`

Reply:
Comes from this commit https://github.com/diffblue/cbmc/commit/d9cc8ac3b4a9a54fa643e5cd7724dd5cc1846d13

Reply:
If I patch the abstract_value.cpp unit test so instead of the default environment, it explicitly makes it top 

```
  auto environment = abstract_environmentt{};
  environment.make_top();
  return environment;
```

then the unit tests pass. @hannes-steffenhagen-diffblue Is this ok with you? If so, I'll fixup that last commit, and we should be good.

Reply:
Sounds reasonable to me.